### PR TITLE
CDAP-6041 fix classloading issues in hive source

### PR DESCRIPTION
**Pull request by albertshau**
Everything should be done using the plugin's classloader as the
context classloader. These hacks can be removed once CDAP-5950 is
fixed.

### CODE REVIEW

File: hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/HiveBatchSource.java [hunk 1]
```diff
@@ -83,38 +82,33 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
 
   @Override
   public void prepareRun(BatchSourceContext context) throws Exception {
-    // This line is to load VersionInfo class here to make it available in the HCatInputFormat.setInput call. This is
-    // needed to support CDAP 3.2 where we were just exposing the classes of the plugin jar and not the resources.
-    LOG.trace("Hadoop version: {}", VersionInfo.getVersion());
```
Comment (bdmogal) [inline hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/HiveBatchSource.java:88]: is it ok to remove this log statement now?

File: hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/HiveBatchSource.java [hunk 2]
```diff
@@ -83,38 +82,33 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
...
-      conf.set(HiveConf.ConfVars.METASTORE_USE_THRIFT_SASL.varname, "true");
-      conf.set("hive.metastore.token.signature", HiveAuthFactory.HS2_CLIENT_TOKEN);
-    }
-    // Use the current thread's classloader to ensure that when setInput is called it can access VersionInfo class
-    // loaded above. This is needed to support CDAP 3.2 where we were just exposing classes to plugin jars and not
-    // resources.
+    // TODO: remove after CDAP-5950 is fixed
     ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
     try {
       Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
```
Comment (bdmogal) [inline hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/HiveBatchSource.java]: nit: seems like by convention, we set the class loader outside the `try...finally`

File: hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/HiveBatchSource.java (hunk 1 above)
Comment (albertshau) [inline hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/HiveBatchSource.java:88]: yep

**Comment by bdmogal** [comment]
couple comments. rest LGTM

**Comment by albertshau** [comment]
closing, this should have been opened against release/1.3

**Pull request closed without merging**